grant-project/grant-monorepo
# User Authentication UI (Pt 2 - Real Users)

### What This Does

Expands on #123 to hook up auth to real users. This makes it so creating an account with a new address actually adds it to the user table, and you are prompted to sign in instead if a user exists with the desired eth address.

### Things to Improve

* There's still no message signing as a part of auth, that will likely be my next PR on top of this.
* Still no persistence with auth state.
* The backend code was quickly thrown together, @dternyak feel free to take a pass at it and either commit directly or throw up another PR.
* I'm leveraging the `TeamMember` type that was made for the create form, but we really ought to consolidate all of these into one unified `User` type. I dropped todos around to be a reminder of this, but I think it's something that we should continue to punt on as some user stuff is in flux.

### Steps to Test

* Create a user, confirm it all worked
* Refresh page, try to auth again, and confirm it pulled in the existing user
* Confirm sign in as that user worked

### Screenshot

<img width="622" alt="screen shot 2018-10-01 at 4 36 12 pm" src="https://user-images.githubusercontent.com/649992/46314302-261dfb00-c598-11e8-99e7-2920d9e482d4.png">
